src/AdoNetCore.AseClient/Internal/Handler/EnvChangeTokenHandler.cs

```diff
@@ -24,10 +24,12 @@ internal class EnvChangeTokenHandler : ITokenHandler
             {"utf8", () => Encoding.UTF8},
         };
         private readonly DbEnvironment _environment;
+        private readonly string _requestedCharset;
 
-        public EnvChangeTokenHandler(DbEnvironment environment)
+        public EnvChangeTokenHandler(DbEnvironment environment, string requestedCharset)
```

Comment:
It took me a little while to understand where the requestedCharset is coming from. Might be better named clientRequestedCharset

Reply:
Fixed